lib/index.js

```diff
@@ -359,80 +217,110 @@ exports.createFunction = async (ctx, functionSpecification) => {
             boundaryId || ctx.body.boundaryId
         }/function/${functionId || ctx.body.functionId}/location`
-    ).set('Authorization', ctx.headers['authorization']); // pass-through authorization
+    ).set('Authorization', `Bearer ${accessToken}`);
 
     return response.body.location;
 };
 
-exports.getStorageClient = (ctx) => {
-    const expiry = 60 * 16; // 15+1 min to align with Lambda lifecycle plus some buffer
-    const accessToken = Jwt.sign({}, Buffer.from(ctx.configuration.fusebit_storage_key, 'base64').toString('utf8'), {
-        algorithm: 'RS256',
-        expiresIn: expiry,
-        audience: ctx.configuration.fusebit_storage_audience,
-        issuer: ctx.configuration.fusebit_storage_issuer_id,
-        subject: ctx.configuration.fusebit_storage_subject,
-        keyid: ctx.configuration.fusebit_storage_key_id,
-        header: { jwtId: Date.now().toString() },
-    });
-
-    const url = generateStorageUrl(ctx.configuration);
+const removeLeadingSlash = (s) => s.replace(/^\/(.+)$/, '$1');
+const removeTrailingSlash = (s) => s.replace(/^(.+)\/$/, '$1');
+
+exports.createStorageClient = async (ctx, accessToken, storageIdPrefix) => {
+    storageIdPrefix = storageIdPrefix ? removeLeadingSlash(removeTrailingSlash(storageIdPrefix)) : '';
+    const functionUrl = Url.parse(ctx.baseUrl);
+    let storageBaseUrl = `${
+            functionUrl.protocol
+        }//${
+            functionUrl.host
+        }/v1/account/${
+            ctx.accountId
+        }/subscription/${
+            ctx.subscriptionId
+        }/storage${
+            storageIdPrefix ? '/' + storageIdPrefix : ''
+        }`;
+
+    const getUrl = (storageSubId) => {
+        storageSubId = storageSubId ? removeTrailingSlash(removeLeadingSlash(storageSubId)) : '';
+        return `${storageBaseUrl}${storageSubId ? ('/' + storageSubId) : ''}`;
+    }
 
-    return {
-        etag: null,
-        expiration: Date.now() + expiry * 1000,
-        get: async function () {
-            const response = await Superagent.get(url)
+    const storageClient = {
+        get: async function (storageSubId) {
+            const response = await Superagent.get(getUrl(storageSubId))
                 .set('Authorization', `Bearer ${accessToken}`)
                 .ok((res) => res.status < 300 || res.status === 404);
-            this.etag = response.body.etag;
-            return response.status === 404 ? undefined : response.body.data;
+            return response.status === 404 ? undefined : response.body;
         },
-        put: async function (data, force) {
-            let request = Superagent.put(url).set('Authorization', `Bearer ${accessToken}`);
-            let payload = { data: data };
-            if (!force) {
-                payload.etag = this.etag;
+        put: async function (data, storageSubId) {
+            await Superagent.put(getUrl(storageSubId))
```

Comment:
Need to return `reponse.body` from this method so folks can grab the etag.

Reply:
Done